### PR TITLE
Refactor `bitrise.yml` and migrate to Unified CI

### PR DESCRIPTION
**Pull request by istvankovacs-bitrise**
### Checklist

- [X] I've read and accepted the [Contribution Guidelines](https://github.com/bitrise-steplib/.github/blob/main/CONTRIBUTING.md)
- Requires _PATCH_ [version update](https://semver.org/)

### Changes
- E2E test cases moved from the root `bitrise.yml` to the `e2e/bitrise.yml`.
- Default workflows (`check` and `sample`) added to the root `bitrise.yml`.

### CODE REVIEW

File: bitrise.yml [hunk 1]
```diff
@@ -47,291 +30,73 @@ workflows:
             #!/bin/bash
             set -e
             bundle exec rspec
-    - trigger-bitrise-workflow:
-        run_if: .IsCI
+
+  e2e_tests:
```
Comment (lpusok) [inline bitrise.yml:34]:
```suggestion
  e2e:
    steps:
    - git::https://github.com/bitrise-steplib/steps-check.git:
        inputs:
        - workflow: e2e
```

File: bitrise.yml [hunk 2]
```diff
@@ -47,291 +30,73 @@ workflows:
...
-        title: Switch working dir to _tmp
-        inputs:
-        - path: $ORIG_BITRISE_SOURCE_DIR/_tmp
-        - is_create_path: "true"
-    - script:
-        title: Cleanup _tmp dir
+            #!/bin/env bash
+            set -ex
+            rm -rf ./_tmp
+    - git::https://github.com/bitrise-steplib/bitrise-step-simple-git-clone.git@master:
```
Comment (lpusok) [inline bitrise.yml:86]:
```suggestion
    - git::https://github.com/bitrise-steplib/bitrise-step-simple-git-clone.git:
```

File: e2e/bitrise.yml [hunk 3]
```diff
@@ -0,0 +1,219 @@
...
+            - content: |-
+                #!/bin/bash
+                set -e
+                echo "BITRISE_EXPORT_METHOD: $BITRISE_EXPORT_METHOD"
+                echo "BITRISE_DEVELOPER_TEAM: $BITRISE_DEVELOPER_TEAM"
+                echo "BITRISE_DEVELOPMENT_CODESIGN_IDENTITY: $BITRISE_DEVELOPMENT_CODESIGN_IDENTITY"
+                echo "BITRISE_DEVELOPMENT_PROFILE: $BITRISE_DEVELOPMENT_PROFILE"
+                echo "BITRISE_PRODUCTION_CODESIGN_IDENTITY: $BITRISE_PRODUCTION_CODESIGN_IDENTITY"
+                echo "BITRISE_PRODUCTION_PROFILE: $BITRISE_PRODUCTION_PROFILE"
+                if [ "$BITRISE_EXPORT_METHOD" != "$DISTRIBUTION_TYPE" ]; then exit 1; fi
```
Comment (lpusok) [inline e2e/bitrise.yml:206]:
```suggestion
                if [ "$BITRISE_EXPORT_METHOD" != "$DISTRIBUTION_TYPE" ]; then exit 1; fi

```

File: e2e/bitrise.yml [hunk 4]
```diff
@@ -0,0 +1,219 @@
...
+
+  _run:
+    steps:
+    - script:
+        inputs:
+        - content: |-
+            #!/bin/env bash
+            set -ex
+            rm -rf ./_tmp
+    - git::https://github.com/bitrise-steplib/bitrise-step-simple-git-clone.git@master:
```
Comment (lpusok) [inline e2e/bitrise.yml:168]:
```suggestion
    - git::https://github.com/bitrise-steplib/bitrise-step-simple-git-clone.git:
```

**Review by lpusok — COMMENTED** [review]
Pls check comments

File: bitrise.yml [hunk 5]
```diff
@@ -47,291 +30,77 @@ workflows:
...
+    steps:
+    - git::https://github.com/bitrise-steplib/steps-check.git:
         inputs:
-        - api_token: "$API_TOKEN"
-        - app_slug: "$APP_SLUG"
-        - workflow_id: test_entitlements
-    - trigger-bitrise-workflow:
-        run_if: .IsCI
+        - workflow: e2e
+    steps:
```
Comment (lpusok) [inline bitrise.yml:39]: pls remove lines 39-70

**Review by lpusok — CHANGES_REQUESTED** [review]

File: e2e/bitrise.yml [hunk 6]
```diff
@@ -0,0 +1,220 @@
...
+  envs:
+  # Shared test secrets
+  - BITRISE_CERTIFICATE_URL_LIST: $BITRISE_CERTIFICATE_URL_LIST
+  - BITRISE_CERTIFICATE_PASSPHRASE_LIST: $BITRISE_CERTIFICATE_PASSPHRASE_LIST
+
+workflows:
+  test_new_certificates:
+    title: Test new Apple Development and Distribution certificates
+    description: |-
+      Test only wired in on Bitrise.io, since this workflow needs to run on new Xcode 11 stack,
```
Comment (lpusok) [inline e2e/bitrise.yml:14]: "Test only wired in..." is no longer true. Do you want to add a check for the Xcode stack in this PR?
Testing with the new CI is not yet in scope is that right?

Comment (istvankovacs-bitrise) [reply e2e/bitrise.yml:14]: Valid point! Previously there was a separate WF with the desired stack to launch this test, but this is obviously not an option for the Unified CI. Do you have something in mind on how to handle this?

Comment (lpusok) [reply e2e/bitrise.yml]: This os OK as short term solution

**Review by lpusok — APPROVED** [review]

**Review by mateherber — APPROVED** [review]

**Pull request merged**